### PR TITLE
Doctrine ORM support

### PR DESCRIPTION
**Pull request by dantleech**
Support for Doctrine ORM routing (as provided by the `RoutingBundle`).

Things to note:
- Locales are not supported (unless we check for the [translatable](https://github.com/Atlantic18/DoctrineExtensions/blob/master/doc/translatable.md) extension).
- Retrieving the content not currently possible (we should add support for this in RoutingBundle).

### CODE REVIEW

File: Adapter/DoctrineOrmAdapter.php [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        $this->entityManager->remove($autoRoute);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function createAutoRoute(UriContext $uriContext, $contentDocument, $autoRouteTag)
+    {
+        $route = new Route();
+        $route->setName(uniqid());
```
Comment (dantleech) [inline Adapter/DoctrineOrmAdapter.php]: Don't really know what the best value for this is, so just used `uniqid` any ideas @dbu @WouterJ  ?

File: Adapter/DoctrineOrmAdapter.php [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        $route->setStaticPrefix($uriContext->getUri());
+
```
Comment (dantleech) [inline Adapter/DoctrineOrmAdapter.php]: Here we need to associate the content with the route. I would suggest that make available an Entity  with `target_type` and `target_identifier` fields in the `RoutingBundle`, somehow.

**Comment by pcabreus** [comment]
Retrieving the content is already posible with ORM. But the AdapterInterface has changed, we can't use the Route entity from RoutingBundle because we need to inherit AutoRouteInterface in our entity. So first we need to create a AutoRoute entity, then tell RoutingBundle that AutoRoute is the base Route.
We also has the problem with the translation, will be supported or not? This could be very diferent comparing with Phpcr Odm implementation.

**Comment by almacbe** [comment]
I think we could close this PR @dantleech @dbu @ElectricMaxxx

**Comment by dbu** [comment]
indeed!
See https://github.com/symfony-cmf/routing-auto-orm-bundle

**Pull request closed without merging**